code/game/machinery/defibrillator_mount.dm

```diff
@@ -51,8 +51,10 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/defibrillator_mount, 28)
 
 	if(!defib)
 		return
-
-	. += "defib"
+	if (defib.icon_state == "defibrusted")
+		. += "defib-rusted"
+	else if (defib.icon_state == "defibunit")
+		. += "defib"
```

Comment:
I feel like it would be a better idea to just have defib units have a var they set to determine their overlay when instead a mount, or to just use the icon_state var itself and rename the states in the overlay icon itself

Reply:
these were the only lines of code for the defib that needed to check icon_state so i decided to not go through the struggle of adding a new var (especially as a new coder)

Reply:
You can also just use the icon state itself, and just rename the icon states in the icon folder to match

Reply:
Although then that might causes issues in the future where people change the name of the icon state but dont update the overlay icon state aswell; so I suppose this is the best solution for now

Reply:
I agree